### PR TITLE
Fix CBMC proofs for socket

### PR DESCRIPTION
**Pull request by tony-josi-aws**
This PR fixes the build issues and proof failures for the Socket related CBMC proofs post the IPv6 changes.

Test Steps
-----------
```
cd test/cbmc/proofs/
python run-cbmc-proofs.py
```
Related Issue
-----------
Failing CBMC proofs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/cbmc/proofs/Socket/vSocketBind/ALLOW_ETHERNET_DRIVER_FILTERS_PACKETS/vSocketBind_harness.c [hunk 1]
```diff
@@ -49,6 +49,13 @@ BaseType_t xApplicationGetRandomNumber( uint32_t * pulNumber )
 
 void harness()
 {
+
+    pxNetworkEndPoints = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
+    __CPROVER_assume( pxNetworkEndPoints != NULL );
+    pxNetworkEndPoints->pxNext = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
+    __CPROVER_assume( pxNetworkEndPoints->pxNext != NULL );
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/Socket/vSocketBind/ALLOW_ETHERNET_DRIVER_FILTERS_PACKETS/vSocketBind_harness.c:56]: We should try and make it more non deterministic.
```suggestion
    if( nondet_bool() )
    {
        pxNetworkEndPoints->pxNext = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
        __CPROVER_assume( pxNetworkEndPoints->pxNext != NULL );
    }
    else
    {
        pxNetworkEndPoints->pxNext = NULL;
    }
```

File: test/cbmc/proofs/Socket/vSocketBind/ALLOW_SOCKET_SEND_WITHOUT_BIND/vSocketBind_harness.c [hunk 2]
```diff
@@ -47,6 +47,13 @@ BaseType_t xApplicationGetRandomNumber( uint32_t * pulNumber )
 
 void harness()
 {
+
+    pxNetworkEndPoints = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
+    __CPROVER_assume( pxNetworkEndPoints != NULL );
+    pxNetworkEndPoints->pxNext = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
+    __CPROVER_assume( pxNetworkEndPoints->pxNext != NULL );
+    pxNetworkEndPoints->pxNext->pxNext = NULL;
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/Socket/vSocketBind/ALLOW_SOCKET_SEND_WITHOUT_BIND/vSocketBind_harness.c:55]: Similarly here too
```suggestion
    if( nondet_bool() )
    {
        pxNetworkEndPoints->pxNext = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
        __CPROVER_assume( pxNetworkEndPoints->pxNext != NULL );
        pxNetworkEndPoints->pxNext->pxNext = NULL;
    }
    else
    {
        pxNetworkEndPoints->pxNext = NULL;
    }
```

File: test/cbmc/proofs/Socket/vSocketBind/DONT_ALLOW_SOCKET_SEND_WITHOUT_BIND/vSocketBind_harness.c [hunk 3]
```diff
@@ -47,6 +47,13 @@ BaseType_t xApplicationGetRandomNumber( uint32_t * pulNumber )
 
 void harness()
 {
+
+    pxNetworkEndPoints = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
+    __CPROVER_assume( pxNetworkEndPoints != NULL );
+    pxNetworkEndPoints->pxNext = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
+    __CPROVER_assume( pxNetworkEndPoints->pxNext != NULL );
+    pxNetworkEndPoints->pxNext->pxNext = NULL;
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/Socket/vSocketBind/DONT_ALLOW_SOCKET_SEND_WITHOUT_BIND/vSocketBind_harness.c:55]: Here similarly we should do this.

File: test/cbmc/proofs/Socket/vSocketWakeUpUser/vSocketWakeUpUser_harness.c [hunk 4]
```diff
@@ -138,7 +145,7 @@ void harness()
     __CPROVER_assume( pxSocket != NULL );
     __CPROVER_assume( pxSocket != FREERTOS_INVALID_SOCKET );
 
-    pxSocket->pxUserWakeCallback = safeMalloc( sizeof( SocketWakeupCallback_t ) );
+    pxSocket->pxUserWakeCallback = &SocketWakeupCallback_Stub;
```
Comment (AniruddhaKanhere) [inline test/cbmc/proofs/Socket/vSocketWakeUpUser/vSocketWakeUpUser_harness.c:148]:
```suggestion
    pxSocket->pxUserWakeCallback = SocketWakeupCallback_Stub; 
```

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]

File: test/cbmc/proofs/Socket/vSocketBind/ALLOW_ETHERNET_DRIVER_FILTERS_PACKETS/vSocketBind_harness.c (hunk 1 above)
Comment (tony-josi-aws) [reply test/cbmc/proofs/Socket/vSocketBind/ALLOW_ETHERNET_DRIVER_FILTERS_PACKETS/vSocketBind_harness.c]: fixed all

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Pull request merged**